### ISSUE TITLE
"Create another token" shortcut problem

### ISSUE THREAD

**Issue by tooomm** [Cockatrice/Cockatrice#2493]
>The keyboard shortcut (Ctrl+G) doesn't work unless the token is created via the token dialog.

Thanks for pointing out @Psithief! https://github.com/Cockatrice/Cockatrice/issues/2453#issuecomment-286949245

**Comment by ZeldaZach** [comment]
That's really awkward how it doesn't just use the same code and has a separate function set...

**Comment by Nightfirecat** [comment]
Sounds like another place to make a function for to prevent that.

**Comment by ZeldaZach** [comment]
I just tested this on Mac and it has the expected work.

```
Right click -> Token menu -> Create token. Ctrl + G creates correct token.
Ctrl + T -> Create token. Ctrl + G creates correct token.
"Token: tokenName" -> Create token. Ctrl + G creates correct token.
```
Did i not test all the cases?

**Comment by Daenyth** [comment]
Are you using a different token for each case so you can tell?

**Comment by tooomm** [comment]
Maybe @Psithief needs to enlighten us here! ;)

**Comment by Psithief** [comment]
It's not what I thought it was!

It's the "all tokens" shortcut. It doesn't overwrite the last token created, which I didn't realise. So, I had used the All Tokens before using the keyboard shortcut, assuming that was a valid test.

tl;dr: Inadvertently tested two variables, blamed the wrong variable for the defect.

**Comment by Daenyth** [comment]
That does raise a question: if "all tokens" makes more than one token, which one should be the "last token"?

Should this even work at all if there's more than one token?

**Comment by Psithief** [comment]
There are only two cards for this edge case that make sense to duplicate all three types:

```
Bestial Menace
Create a 1/1 green Snake creature token, a 2/2 green Wolf creature token, and a 3/3 green Elephant creature token.
```

```
Trostani's Summoner
When Trostani's Summoner enters the battlefield, create a 2/2 white Knight creature token with vigilance, a 3/3 green Centaur creature token, and a 4/4 green Rhino creature token with trample.
```

It doesn't make any sense to do so for Planeswalkers that create emblems and tokens, but the All Tokens function shouldn't work for them in the first place. (Maybe an exclusion flag is needed here.)

**Comment by Daenyth** [comment]
> (Maybe an exclusion flag is needed here.)

Keep it simple. Let's not tie in too much game knowledge. We can do one menu entry per token + 1 create all, and the user can just use it or not when it makes sense

**Comment by ZeldaZach** [comment]
What should we do with this?

**Comment by ctrlaltca** [comment]
The "create all tokens" overwrites the "last token" only when a single token is created:

https://github.com/Cockatrice/Cockatrice/blob/master/cockatrice/src/player.cpp#L1131
```
    /*
     * If we made a token via "Token: TokenName"
     * then lets allow it to be created via create another
     */
```
The "create another token" menu options also seems to be greyed out correctly.

Back to …@daenyth's question:

> That does raise a question: if "all tokens" makes more than one token, which one should be the "last token"? 
> Should this even work at all if there's more than one token?

Imho: no

**Comment by tooomm** [comment]
I guess players expect it to work for several token creations at once too. Even though the menu says "last token".

Keep in mind that with https://github.com/Cockatrice/Cockatrice/issues/2262 we are going to have way more cards which want to create several tokens at once! So this gets really handy.
But then the menu should be something like `Create last used token(s)` in my opinion
(or without "used", or with "Re-Create").

**Comment by ebbit1q** [comment]
This only doesn't work for creating copies of cards for me, it does work when you create a related card though. Considering you'd most often create tokens that way it seems fine. However it only creates a single token even though the card created multiple.
